### PR TITLE
test: complete Windows zip archive content validation from POC

### PR DESCRIPTION
**Pull request by nr-ahemsath**
## Description

This PR completes the POC work for doing Windows ZIP archive content validation.  I liked @nrcventura's approach used for the MSI better, so I ended up doing that instead.  This created some code duplication so I refactored the common code into a helper class.

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: build/ArtifactBuilder/Artifacts/ZipArchive.cs [hunk 1]
```diff
@@ -56,73 +55,73 @@ protected override void InternalBuild()
...
-            // 2. Why do the RootInstallDirectoryComponents differ between framework and core?
-            // 3. How to account for "agentInfo.json" in the home dir (both framework and core)?
+            ValidationHelpers.AddFilesToCollectionWithNewPath(expectedComponents, installedFilesRoot, _frameworkAgentComponents.RootInstallDirectoryComponents);
 
-            // cleanup
-            FileHelpers.DeleteDirectories(unpackDir);
-        }
+            var netframeworkFolder = Path.Join(installedFilesRoot, FrameworkSubDirectoryName);
+            ValidationHelpers.AddFilesToCollectionWithNewPath(expectedComponents, netframeworkFolder, _frameworkAgentComponents.AgentHomeDirComponents);
+            expectedComponents.Add(Path.Join(netframeworkFolder, AgentInfo.AgentInfoFilename));
```
Comment (nrcventura) [inline build/ArtifactBuilder/Artifacts/ZipArchive.cs:105]: You might not need to special case the agentinfo filename. The file is already available as a property on the frameworkcomponents class. Alternatively, the agentinfo json file is part of the ConfigurationComponents property too.

File: build/ArtifactBuilder/Artifacts/ZipArchive.cs [hunk 2]
```diff
@@ -56,73 +55,73 @@ protected override void InternalBuild()
...
+                throw new PackagingException($"The unpacked ZIP archive was missing the expected component {missingComponent}");
+            }
 
-            // core agent
+            var unexpectedUnpackedComponents = new SortedSet<string>(unpackedComponents, StringComparer.OrdinalIgnoreCase);
+            unexpectedUnpackedComponents.ExceptWith(expectedComponents);
+            foreach (var unexpectedComponent in unexpectedUnpackedComponents)
+            {
+                throw new PackagingException($"The unpacked ZIP archive contained an unexpected component {unexpectedComponent}");
+            }
```
Comment (nrcventura) [inline build/ArtifactBuilder/Artifacts/ZipArchive.cs:89]: We might be able to extract this code into the helper too, if we can parameterize the part of the exception messages.

File: build/ArtifactBuilder/Artifacts/ZipArchive.cs (hunk 1 above)
Comment (nr-ahemsath) [reply build/ArtifactBuilder/Artifacts/ZipArchive.cs]: Thanks, that works better.

File: build/ArtifactBuilder/Artifacts/ZipArchive.cs (hunk 2 above)
Comment (nr-ahemsath) [reply build/ArtifactBuilder/Artifacts/ZipArchive.cs]: Good idea.

**Pull request merged**